apps/docs/content/guides/realtime/authorization.mdx

```diff
@@ -0,0 +1,288 @@
+
+</Admonition>
+
+## How it works
+
+Realtime uses tables in your database's `realtime` schema to track users and Channel topics. By writing RLS policies against these tables, you can control the access users have to a Channel topic and features within a Channel topic.
+
+The validation is done when the user connects. When their WebSocket connection is established, their permissions are calculated based on:
+
+- The RLS policies on tables in your `realtime` schema
+- The user information sent as part of their [Auth JWT](/docs/guides/auth/jwts)
+- The request headers
+- The Channel topic the user is trying to connect to
+
+<Admonition type="caution">
+
+Increased RLS complexity can impact database performance and connection time, leading to higher connection latency and decreased join rates.
+
+</Admonition>
+
+## Helper functions
+
+You can use the following helper functions when writing RLS policies:
+
+### `realtime.topic`
+
+Returns the Channel topic the user is attempting to connect to.
+
+```sql
+create policy "authenticated users can read all messages in room_1"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+  (select realtime.topic()) = 'room_1'
+);
+```
+
+## Examples
+
+The following examples use this schema:
+
+```sql
+create table public.profiles (
+  email text not null,
+  user_id uuid references auth.users (id)
+);
+
+create table public.rooms_users (
+  topic text references realtime.messages (topic),
+  user_id uuid references auth.users (id),
+  created_at timestamptz default current_timestamp
+);
+
+// Trigger to populate profiles table
+create or replace function insert_user () returns trigger as
+$$
+  begin
+    insert into public.profiles (user_id, email) values (new.id, new.email);
+    return new;
+  end;
+$$ language plpgsql;
+
+create or replace trigger "on_new_auth_create_profile"
+after insert on auth.users for each row
+execute function insert_user ();
+
+revoke execute on function insert_user () from public, anon, authenticated;
+grant execute on function insert_user () TO supabase_auth_admin;
+grant insert on table public.profiles to supabase_auth_admin;
+```
+
+### Broadcast
+
+The `extension` field on the `realtime.messages` table records the message type. For Broadcast messages, the value of `realtime.messages.extension` is `broadcast`. You can check for this in your RLS policies.
+
+#### Allow a user to join (and read) a Broadcast topic
+
+To join a Broadcast Channel, a user must have at least one read or write permission on the Channel topic.
+
+Here, we allow reads (`select`s) for users who are linked to the requested topic within the relationship table `public.room_users`:
+
+```sql
+create policy "authenticated users can join topic"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
```

Comment:
I wrapped all of these helper function calls in sub-selects because we recommend that in Performance Advisor, if that doesn't apply here, feel free to revert

Reply:
nice one, thanks @charislam!